lighthouse-core/test/config/budget-test.js

```diff
@@ -197,6 +197,54 @@ describe('Budget', () => {
     });
   });
 
+  describe('budget matching', () => {
+    const budgets = [{
+      path: '/',
+      resourceSizes: [
+        {
+          resourceType: 'script',
+          budget: 0,
+        },
+      ],
+    },
+    {
+      path: '/file.html',
+      resourceSizes: [
+        {
+          resourceType: 'image',
+          budget: 0,
+        },
+      ],
+    },
+    {
+      path: '/not-a-match',
+      resourceSizes: [
+        {
+          resourceType: 'document',
+          budget: 0,
+        },
+      ],
+    },
+    ];
+    it('returns the last matching budget', () => {
+      const budget = Budget.matchingBudget(budgets, 'http://example.com/file.html');
+      expect(budget).toEqual(budgets[1]);
+    });
+
+    it('does not mutate the budget config', async () => {
+      const configBefore = JSON.parse(JSON.stringify(budgets));
+      Budget.matchingBudget(configBefore, 'https://example.com');
+      const configAfter = JSON.parse(JSON.stringify(budgets));
+      expect(configBefore).toEqual(configAfter);
+    });
+
+    it('returns a copy of the matching budget', () => {
```

Comment:
I added a comment to ask for this test before I noticed it, ❤️ it! nice job :)

Reply:
> returns a copy of the matching budget

Why is this important, though? @khempenius you called it out as a fix, but I thought the shallow clone was just used to make `reverse()` not mutate the original, not as a guarantee that the returned thing is a copy.